docs/articles/documentation/test-api/selecting-page-elements/examples-of-working-with-dom-elements.md

```diff
@@ -233,6 +234,77 @@ test('My test', async t => {
 });
 ```
 
+## Select Elements With Dynamic IDs
+
+TestCafe selectors should use element identifiers that persist between test runs. However, many JavaScript frameworks generate dynamic IDs for page elements. To identify elements whose `id` attribute changes, use selectors based on the element's class, content, tag name, or position:
+
+* [withText](selectors/functional-style-selectors.md#withtext),
+* [withExactText](selectors/functional-style-selectors.md#withexacttext),
+* [withAttribute](selectors/functional-style-selectors.md#withattribute),
+* [parent](selectors/functional-style-selectors.md#parent),
+* [child](selectors/functional-style-selectors.md#child),
+* [sibling](selectors/functional-style-selectors.md#sibling),
+* [nextSibling](selectors/functional-style-selectors.md#nextsibling),
+* [prevSibling](selectors/functional-style-selectors.md#prevsibling).
+
+**Example**
+
+```html
+<html>
+    <body>
+        <div id="j9dk399sd304" class="container">
+            <div id="dsf054k45o3e">Item 1</div>
+            <div id="lk94km904wfv">Item 2</div>
+        </div>
+    </body>
+</html>
+```
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `http://localhost/`;
+
+test('My test', async t => {
+    const container = Selector('div').withAttribute('class', 'container');
+    const item1     = Selector('div').withText('Item 1');
+    const item2     = container.child(1);
+});
+```
+
+If the element's ID is partially dynamic, you can use the following CSS selectors to match the ID's static part:
+
+* [[attribute~=value]](https://www.w3schools.com/cssref/sel_attribute_value_contains.asp),
```

Comment:
we also have `withAttribute` method that supports RegExp arguments for attribute's name and value, e.g.
`await t.click(Selector('button').withAttribute('id', /some_known_part\d+_test2/))`

Reply:
Maybe we shouldn't recommend CSS selectors at all in this case?

Reply:
I think it's ok if we mention both variants